Scheduler: Add snapToCellsMode API Desc. (#8599)
DevExpress/devextreme-documentation: api-reference/10 UI Components/dxScheduler/1 Configuration/snapToCellsMode.md

---
---
##### shortDescription

> Specifies how Scheduler snaps appointment edges to the cell grid when loading data (all views).

Reviewer: ```suggestion
Specifies how the Scheduler component snaps appointments to grid (applies to all view types).
```

---

> If a Scheduler is bound to a [dataSource](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#dataSource), it can snap appointments to grid. If this functionality is active, the Scheduler increases appointment dimensions (never shrinks appointments).

Reviewer: ```suggestion
If a Scheduler is bound to a [dataSource](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#dataSource), you can choose whether the control snaps appointments to grid. If this functionality is active, the Scheduler increases appointment dimensions (never shrinks appointments).
```

> Scheduler supports snapping appointments loaded from the component [dataSource](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#dataSource) to the cell grid. Scheduler increases appointment dimensions to implement this capability. The component does not shrink appointment containers to snap to the cell grid.

Reviewer: ```suggestion
If a Scheduler is bound to a [dataSource](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#dataSource), it can snap appointments to grid. If this functionality is active, the Scheduler increases appointment dimensions (never shrinks appointments).
```

The following **snapToCellsMode** values are available:

> - *"always"*: Appointments occupy cells fully.

Reviewer: ```suggestion
- *"always"*: Scheduler snaps all appointments to grid.
```

> - *"auto"*: Appointments that cover more than half of a cell are stretched to occupy the full cell.

Reviewer: ```suggestion
- *"auto"*: Appointments that cover more than half of a cell are stretched to occupy the entire cell.
```

> - *"never"*: Scheduler does not increase appointment dimensions.

Reviewer: ```suggestion
- *"never"*: Scheduler does not snap appointments to grid.
```

---
---
##### shortDescription
Specifies how the Scheduler component snaps appointments to the grid (applies to all view types).

---
The Scheduler can snap appointments to the grid. When this feature is enabled, the component increases appointment dimensions and never reduces them.

The following **snapToCellsMode** values are available:

- *"always"*: Scheduler snaps all appointments to the grid.
- *"auto"*: Only appointments that occupy two or fewer cells are snapped to the grid.
- *"never"*: Scheduler does not snap appointments to the grid.

The default **snapToCellsMode** value varies by view:

<table class="dx-table">
    <tr>
        <th>Value</th>
        <th>Scheduler Views</th>
    </tr>
    <tr>
        <td><i>"never"</i></td>
        <td><i>"day"</i>, <i>"week"</i>, <i>"workWeek"</i>, <i>"timelineDay"</i>, <i>"timelineWeek"</i>, <i>"timelineWorkWeek"</i></td>
    </tr>
    <tr>
        <td><i>"always"</i></td>
        <td><i>"month"</i>, <i>"timelineMonth"</i></td>
    </tr>
</table>

[note]

> - **snapToCellsMode** does not affect appointment [resizing](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/editing/#allowResizing) behavior.

Reviewer 2: ```suggestion
- **snapToCellsMode** does not affect appointment [resizing](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/editing/#allowResizing).
```
